registry/storage/driver/storagedriver.go

```diff
@@ -103,7 +103,7 @@ type FileWriter interface {
 	Size() int64
 
 	// Cancel removes any written content from this FileWriter.
-	Cancel() error
+	Cancel(context.Context) error
```

Comment:
@milosgajdos fair enough, I replaced yaml roundtrip with mapstructure.

Regarding the FileWriter interface, @corhere correctly pointed out that since Cancel is invoked out of band, we cannot use the original context in the event that the cancelation api itself requires a context.  A context is available to callers of Cancel, so it made sense to extend the interface.

In fact the swift driver had this problem, and it used context.Background() as a workaround which isn't ideal.  I could have done the same here, but it felt wrong.

Reply:
Gotchya. Makes sense.